packages/dataviews/src/filter-and-sort-data-view.ts

```diff
@@ -142,6 +142,16 @@ export function filterSortAndPaginate< Item >(
 			filteredData.sort( ( a, b ) => {
 				const valueA = fieldToSort.getValue( { item: a } ) ?? '';
 				const valueB = fieldToSort.getValue( { item: b } ) ?? '';
+
+				if (
+					typeof valueA === 'number' &&
+					typeof valueB === 'number'
```

Comment:
This ideally should come from the "field" type and not by guessing the value type. Any other words, the "field types" should define a sorting function. But this is more down the road. This is ok for now.